lib/JSX.ml

```diff
@@ -0,0 +1,112 @@
+module Attribute = struct
+  (** Used internally, no need to use *)
+
+  type t =
+    | Bool of (string * bool)
+    | String of (string * string)
+    | Style of string
+    | Event of (string * string)
+
+  let t_to_string attr =
+    match attr with
+    (* false attributes don't get rendered *)
+    | Bool (_, false) -> ""
+    (* true attributes render solely the attribute name *)
+    | Bool (k, true) -> k
+    | Style styles -> Printf.sprintf "style=\"%s\"" styles
+    | Event (name, value) -> Printf.sprintf "%s=\"%s\"" name value
+    | String (k, v) -> Printf.sprintf "%s=\"%s\"" k (Html.encode v)
+
+  let to_string attrs =
+    match List.map t_to_string attrs with
+    | [] -> ""
+    | rest -> " " ^ (rest |> List.rev |> String.concat " " |> String.trim)
+end
+
+type node = {
+  tag : string;
+  attributes : Attribute.t list;
+  children : element list;
+}
+
+and element =
+  | Null
+  | String of string
+  | Unsafe of string (* text without encoding *)
+  | Fragment of element list
+  | Node of node
+  | Component of (unit -> element) (* TODO: Is this unused? *) [@warning "-37"]
+  | List of element list
+
+let string txt = String txt
```

Comment:
I would keep this as `text`, now that we have `to_string`

Reply:
I don't really mind which one, but it was sprung out from the accumulation of changes when switching between `html_of_jsx` and React. 

E.g. 
1. `s/JSX./React./g` 
2. `s/React.text/React.string/g` 
3. ... and so forth

If you want `text` we can go for it, but if the issue is the lack of distinction between `string` and `to_string` then could also consider replacing `to_string` with `render`, but your call! 🤷‍♂️

Reply:
string/render make sense indeed